docs/project/list-of-diagnostics.md

```diff
@@ -270,3 +270,5 @@ The diagnostic id values reserved for .NET Libraries analyzer warnings are `SYSL
 | Suppression ID           | Suppressed Diagnostic ID | Description |
 | :----------------------- | :----------------------- | :---------- |
 | __`SYSLIBSUPPRESS0001`__ | CA1822                   | Do not offer to make methods static when the methods need to be instance methods for a custom marshaller shape. |
+| __`SYSLIBSUPPRESS0002`__ | IL2026                   | ConfigurationBindingGenerator: suppress RequiresUnreferencedCode diagnostic for binding call that has been intercepted by a generated static variant. |
```

Comment:
Is this needed because we no longer binding to a generated extension method?

Reply:
This is a result of https://github.com/dotnet/roslyn/issues/68669.

The trimming analyzer cannot detect if a call has been intercepted and execute it's analysis on the interceptor so it ends up firing false positives.

The approach we're going with for .NET 8 is shipping a suppressor alongside the generator to suppress the false positives if we can verify that the generator produced an interceptor.